spec/controllers/posts_controller_spec.rb

```diff
@@ -1052,6 +1052,85 @@
       end
     end
 
+    it "gives the correct previous post" do
+      user = create(:user)
+      board = create(:board, creator: user)
+      section = create(:board_section, board: board)
+      extra = create(:post, user: user, board: board, section: section)
+      prev = create(:post, user: user, board: board, section: section)
+      hidden = create(:post, board: board, section: section, privacy: Concealable::PRIVATE)
+      post = create(:post, user: user, board: board, section: section)
+      extra.update!(section_order: 0)
+      prev.update!(section_order: 1)
+      hidden.update!(section_order: 2)
+      post.update!(section_order: 3)
+
+      get :show, params: { id: post.id }
+
+      expect(assigns(:prev_post)).to eq(prev)
+    end
+
+    it "gives the correct next post" do
+      user = create(:user)
+      board = create(:board, creator: user)
+      section = create(:board_section, board: board)
+      post = create(:post, user: user, board: board, section: section)
+      hidden = create(:post, board: board, section: section, privacy: Concealable::PRIVATE)
+      nextp = create(:post, user: user, board: board, section: section)
+      extra = create(:post, user: user, board: board, section: section)
+      post.update!(section_order: 0)
+      hidden.update!(section_order: 1)
+      nextp.update!(section_order: 2)
+      extra.update!(section_order: 3)
+
+      get :show, params: { id: post.id }
+
+      expect(assigns(:next_post)).to eq(nextp)
+    end
+
+    it "does not give previous with only a non-visible post in section" do
+      user = create(:user)
+      board = create(:board, creator: user)
+      section = create(:board_section, board: board)
+      hidden = create(:post, board: board, section: section, privacy: Concealable::PRIVATE)
+      post = create(:post, user: user, board: board, section: section)
+      hidden.update!(section_order: 0)
+      post.update!(section_order: 1)
+
+      get :show, params: { id: post.id }
+
+      expect(assigns(:prev_post)).to be_nil
+    end
+
+    it "does not give next with only a non-visible post in section" do
+      user = create(:user)
+      board = create(:board, creator: user)
+      section = create(:board_section, board: board)
+      post = create(:post, user: user, board: board, section: section)
+      hidden = create(:post, board: board, section: section, privacy: Concealable::PRIVATE)
+      post.update!(section_order: 0)
+      hidden.update!(section_order: 1)
+
+      get :show, params: { id: post.id }
+
+      expect(assigns(:next_post)).to be_nil
+    end
+
+    it "handles very large mostly-hidden sections correctly" do
```

Comment:
Maybe "as expected" rather than "correctly" (since in a perfect world this would handle arbitrary numbers of hidden posts).